### PR TITLE
permit global augmentations to introduce new names

### LINKED ISSUE #8102: Allow `declare global` to declare new names
```ts

export {};

declare global {
    interface SymbolConstructor {
        observable: symbol;
    }
}

Symbol.observable;
```

this is an error if ES6 library is not included, but not otherwise. most users would find this confusing.

### PR DESCRIPTION
**Pull request by vladima**
fixes #8102

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -15596,6 +15596,9 @@ namespace ts {
                 case SyntaxKind.InterfaceDeclaration:
                 case SyntaxKind.ModuleDeclaration:
                 case SyntaxKind.TypeAliasDeclaration:
```
Comment (DanielRosenwasser) [inline src/compiler/checker.ts]: There's also a case for an `import =` declaration that reports this error in global module augmentations.

**Comment by vladima** [comment]
0f323ea also fixes asymmetry that we currently have: in ambient namespaces without explicitly exported declarations all entities were considered exported except imports. This is technically a breaking change in following scenario however I have no idea what this code might do in the first place 

```ts
declare namespace A.B {
	let x: number;
}

declare namespace M {
	import X = A.B.x;
	let X: string;
}

M.X // refers to let X: string
```

**Comment by mhegazy** [comment]
Can you run definitely typed tests on this change and see if there are any new errors? i have definitely seen this pattern in the past.

**Comment by vladima** [comment]
revering back since it breaks the following case:
```ts
declare module "A" {
    import X = A.B.C
    let y = X.M1;
}
declare module "A" {
    import X = A.B.C
    let y = X.M2;
}```

**Comment by vladima** [comment]
@mhegazy any other comments?

**Comment by mhegazy** [comment]
:+1:

**Pull request merged**